index.js

```diff
@@ -95,6 +96,14 @@ Promise.resolve()
           console.warn(chalk.bold.cyan('Waiting for file changes...'))
         })
         .on('change', file => {
+          if (argv.replace) {
```

Comment:
Isn't that equal to just `if (argv.replace) return`? 
Could you provide a small usage example/description please ?

Does this really trigger a rebuild of the replaced files so you enter a 'replace loop'  within the same `process` 
when watch mode is enabled, or should this just falsely prevent the CLI to not throw an `{Error}` if `--replace --watch` are used together so the file contents are replaced **once** but without exiting the `process` ?

Reply:
No, it is not the same. This change makes postcss-cli smart enough to only trigger compiles for external changes in `--replace` mode.

The reason I need this is that babel is processing my css files from src -> lib and then postcss is processing the files in place. With this change, I can run `babel --watch` and then `postcss --watch --replace`. Then when I modify a css file in src it is processed by babel and written to lib. Postcss detects the change in lib and processes it in place.

Reply:
Please define external changes in more detail (when babel copies the files from `src` => `lib` ?)

> The reason I need this is that babel is processing my css files from src -> lib and then postcss is processing the files in place. With this change, I can run babel --watch and then postcss --watch --replace. Then when I modify a css file in src it is processed by babel and written to lib. Postcss detects the change in lib and processes it in place.

That's what I mean, if you would use `postcss --replace --watch` in e.g the `src` folder without anything else changing the files (standalone), `postcss-cli` would process the files once but keeps the `process` alive without transforming the replaced files again, making it a necessity to kill the `process` manually. So for standalone usage this is wrong.... Can you provide your build scripts ? Why is `npm-run-all` with `babel src -d lib --watch` && `postcss src/*.css --dir lib --watch?` not sufficient ?

Reply:
My process looks something like this. 

``` shell
rm -rf lib/*

babel src --out-dir lib --source-maps --copy-files

postcss lib/**/*.css --replace --map --watch

# Alternatively, we could call babel with the --watch option above and poll lib/**/*.css until it's count matches src/**/*.css, and then run postcss
babel src --out-dir lib --source-maps --copy-files --watch
```

Regarding `postcss --replace --watch` doing nothing, before this change it just continuously rebuilds all css files until a user manually kills the process. With this change it builds all css files and then waits for a change.

![image](https://user-images.githubusercontent.com/647452/33457937-013c2c3a-d5d9-11e7-9a1a-3f34cd41b797.png)

Reply:
If it helps, we could tweak the "Ignoring change to busy file" message to a note about using `--watch` with `--replace`.

Reply:
Consider using https://github.com/mysticatea/npm-run-all/blob/master/docs/npm-run-all.md for that purpose instead please 

```
npm i -D npm-run-all
```

**package.json**
```js
{
   "scripts": {
        "clean": "rimraf lib/*",
        "css": "postcss src/*.css --dir lib --watch",
        "js": "babel src/*.js --out-dir lib --watch",
        "start:" "npm-run-all clean css js"
    },
   "devDependencies": {
        "npm-run-all": "^x.x.x"
         ...
    }
}
```

postcss-cli shouldn't rely on third-party behaviour in any way

Reply:
I'm sorry, I should have mentioned that I'm running babel and postcss asynchronously. This is closer to what I'm running, aside from some error handling logic.

``` shell
const {spawn} = require('child-process-promise')

async function transpile() {
    await spawn('rm', ['-rf', 'lib'])

    await spawn('babel-cli/bin/babel', [
        'src', '--out-dir', 'lib', '--source-maps', '--copy-files',
    ], {stdio: 'inherit'})
    
    spawn('postcss-cli/bin/postcss', [
        'lib/**/*.css', '--replace', '--map', '--watch',
    ], {stdio: 'inherit'})
    
    spawn('babel-cli/bin/babel', [
        'src', '--out-dir', 'lib', '--source-maps', '--copy-files',
    ], {stdio: 'inherit'})
}
transpile()
```

Your suggestion to use npm-run-all doesn't work, postcss still triggers its own watch and recompiles over and over until I kill it.

Reply:
I'm still not convinced that the coupling between `babel` && `postcss` is necessary. Why does 

```
spawn('postcss-cli/bin/postcss', [
        'lib/**/*.css', '--dir, 'lib', '--map', '--watch',
    ], {stdio: 'inherit'})
```

not work ? The underlying concept of e.g `npm-run-all` will also likely be to spawn `child_processes` and handle them. There must be a way to handle this outside of `postcss-cli` in within your custom logic somehow

Reply:
That doesn't work because I need babel to process my css before postcss. I don't understand why you need me to justify that need, but that's not changing. If you need me to expound further on why I can, but I'm not sure that's really helpful.

I'm also not the only one with this use case, a number of other users have asked for it in on issue #126.

Reply:
Folks in #126 want  `--watch` to work with the **replaced** files in the same directory, your proposed solution wouldn't enable that use case.

> That doesn't work because I need babel to process my css before postcss.

Why?  I assumend `--copy-files` just cp your css atm, so yes please show your `.babelrc`

What about ? 

```js
spawn('babel-cli/bin/babel', [
        'src', '--out-dir', 'lib', '--source-maps', '--copy-files',
    ], {stdio: 'inherit'})
    .then(() => {
      spawn('postcss-cli/bin/postcss', [
          'lib/**/*.css', '--replace', '--map',
      ], {stdio: 'inherit'})
      
      return null
    })
    .catch((err) => err)
```

We really can't accept this PR in it's current state as it introduces a bug for simple `postcss src/*.css --replace` usage.

Reply:
I'm using babel-plugin-css-modules-transform to transform the css classnames in my javascript and css code. I've tried using a couple postcss plugins to do the css modules transforms, and none of them work quite right.

As for your suggestion, I missed the `--watch` option on my last snippet, it should have been.

``` javascript
const {spawn} = require('child-process-promise')

async function transpile() {
    await spawn('rm', ['-rf', 'lib'])

    await spawn('babel-cli/bin/babel', [
        'src', '--out-dir', 'lib', '--source-maps', '--copy-files',
    ], {stdio: 'inherit'})
    
    spawn('postcss-cli/bin/postcss', [
        'lib/**/*.css', '--replace', '--map', '--watch',
    ], {stdio: 'inherit'})
    
    spawn('babel-cli/bin/babel', [
        'src', '--out-dir', 'lib', '--source-maps', '--copy-files', '--watch',
    ], {stdio: 'inherit'})
}
transpile()
```

And your promise version is the same as my async/await version, but without the watch. Without `--watch` it works great, but when I add `--watch` I get the same infinite loop that users complain about in #126. Speaking of #126, I've read over the comments again, and it does seem like my change would fix their issue, but maybe I'm missing something.

What is the bug for simple `postcss src/*.css --replace` usage that you mention?